seissolxdmfwriter/seissolxdmfwriter/seissolxdmfwriter.py

```diff
@@ -0,0 +1,80 @@
+import h5py
+
+
+def write_seissol_xdmf(
+    prefix, nNodes, nCells, lDataName, dt, node_per_element, lidt, precision
+):
+    prec = 8 if precision == "double" else 4
+    topology = "Tetrahedron" if node_per_element == 4 else "Triangle"
+    xdmf = """<?xml version="1.0" ?>
+<!DOCTYPE Xdmf SYSTEM "Xdmf.dtd" []>
+<Xdmf Version="2.0">
+ <Domain>"""
+    for i, idt in enumerate(lidt):
+        xdmf += f"""
+  <Grid Name="TimeSeries" GridType="Collection" CollectionType="Temporal">
+   <Grid Name="step_{idt}" GridType="Uniform">
+    <Topology TopologyType="{topology}" NumberOfElements="{nCells}">
+     <DataItem NumberType="Int" Precision="8" Format="HDF" Dimensions="{nCells} {node_per_element}">{prefix}.h5:/mesh0/connect</DataItem>
+    </Topology>
+    <Geometry name="geo" GeometryType="XYZ" NumberOfElements="{nNodes}">
+     <DataItem NumberType="Float" Precision="8" Format="HDF" Dimensions="{nNodes} 3">{prefix}.h5:/mesh0/geometry</DataItem>
+    </Geometry>
+    <Time Value="{idt*dt}"/>"""
+        for dataName in lDataName:
+            xdmf += f"""
+    <Attribute Name="{dataName}" Center="Cell">
+     <DataItem ItemType="HyperSlab" Dimensions="{nCells}">
+      <DataItem NumberType="UInt" Precision="4" Format="XML" Dimensions="3 2">{i} 0 1 1 1 {nCells}</DataItem>
+      <DataItem NumberType="Float" Precision="{prec}" Format="HDF" Dimensions="{i+1} {nCells}">{prefix}.h5:/mesh0/{dataName}</DataItem>
+     </DataItem>
+    </Attribute>"""
+        xdmf += """
+   </Grid>
+  </Grid>"""
+    xdmf += """
+ </Domain>
+</Xdmf>
+"""
+    with open(prefix + ".xdmf", "w") as fid:
+        fid.write(xdmf)
+    print(f"done writing {prefix}.xdmf")
+
+
+def write_seissol_h5(prefix, lDataName, xyz, connect, lData, lidt, precision):
+    myDtype = "float64" if precision == "double" else "float32"
+    nCells, node_per_element = connect.shape
+    with h5py.File(prefix + ".h5", "w") as h5f:
+        h5f.create_dataset("mesh0/connect", (nCells, node_per_element), dtype="i8")
+        h5f["mesh0/connect"][:, :] = connect[:, :]
+        h5f.create_dataset("mesh0/geometry", xyz.shape, dtype="d")
+        h5f["mesh0/geometry"][:, :] = xyz[:, :]
+        for k, dataName in enumerate(lDataName):
+            hdname = "mesh0/" + dataName
+            h5f.create_dataset(hdname, (len(lidt), nCells), dtype=myDtype)
```

Comment:
I just wanted to write integer data. I propose to remove `precision` and add
```
dtypeDict = {"int64": "i8", "int32": "i4", "float64": "float64", "float32": "float32"}
...
    h5f.create_dataset(hdname, (len(lidt), nCells), dtype=dtypeDict[lData[k].dtype.name])
```

Reply:
I've integrated your changes